stingray/tests/test_events.py

```diff
@@ -268,6 +268,16 @@ def test_event_file_read(self, fname):
         fname = os.path.join(datadir, fname)
         EventList.read(fname, fmt="hea")
 
+    def test_event_file_read_laxpc(self):
+        """Test LAXPC file reading"""
+        fname = os.path.join(datadir, "laxpc_file_read.fits")
+        with pytest.warns(UserWarning, match="HDU EVENTS"):
+            with pytest.warns(UserWarning, match="No valid GTI"):
+                ev = EventList.read(fname, fmt="hea")
+        assert ev.instr.lower() == "laxpc3"
+        assert ev.mission.lower() == "astrosat"
+        assert ((ev.filter_detector_id(detector_id=1)).main_array_length) == 453
```

Comment:
```suggestion
        assert ((ev.filter_detector_id(detector_id=1)).main_array_length) == 453
        assert ((ev.filter_detector_id(detector_id=[1])).main_array_length) == 453
```

This should also take care of the array case